plugins/cq_cache/cq_cache.py

```diff
@@ -74,24 +76,19 @@ def build_file_name(fct, *args, **kwargs):
     Returns a file name given the specified function and args.
     If the function and the args are the same this function returns the same filename
     """
-    SPACER = "_"
-    file_name = fct.__name__
-    for arg in args:
-        if isinstance(arg, cq.Workplane):
-            raise TypeError(
-                "Can not cache a function that accepts Workplane objects as argument"
-            )
-        file_name += SPACER + str(hash(arg))
-    for kwarg_value in kwargs.values():
-        if isinstance(kwarg_value, cq.Workplane):
-            raise TypeError(
-                "Can not cache a function that accepts Workplane objects as argument"
-            )
-        file_name += SPACER + str(hash(kwarg_value))
-    file_name = bytes(file_name, "utf-8")
-    return base64.urlsafe_b64encode(file_name).decode(
-        "utf-8"
-    )  # compacts the long string of hash ints into a urlsafe string
+    if cq.Workplane in (type(x) for x in chain(args, kwargs.values())):
+        raise TypeError(
+            "Can not cache a function that accepts Workplane objects as argument"
+        )
+
+    # hash all relevant variables
+    hasher = hashlib.md5()
+    for val in [fct.__name__, args.__repr__(), kwargs.__repr__()]:
```

Comment:
How is this different from my initial way of doing it using `str(arg)` ? 
From what I understand the difference with my initial code is that now instead of storing the `str(arg)` directly in the file name, you concatene everything is a resulting hash str. What for ?

Reply:
> How is this different from my initial way of doing it using `str(arg)` ?

Yeah, not much difference. I had thought I could get it much more accurate, but the `hash` implementation is mainly for supporting dicts and isn't suitable. 

It's possible to hash the bytes output from pickle. This would be as accurate as I was hoping for, but you can't pickle most CQ types, so it isn't practical for us.

`repr` is slightly better than `str`, since:
```
>>> str(1) == str("1")
True
>>> repr(1) == repr("1")
False
```

I haven't solved the main problems with a `str(arg)` based caching though, mainly that any complicated object (including lambdas, generators, `cq.Plane` and `cq.Location`) will never compare the same and will be a seperate cache value each time the function is called. :shrug: 

> From what I understand the difference with my initial code is that now instead of storing the `str(arg)` directly in the file name, you concatene everything is a resulting hash str. What for ?

The hash string is a fixed length, this could easily blow out past filename limits with large args. Especially given it's going to be used on computationally heavy functions where you wouldn't be surprised to see a large array of (x, y, z) tuples as an argument.

Reply:
Alright thanks for the explanation. It's a bit unfortunate that we can't support complicated types but maybe someone will have a better idea about this in the future !

Reply:
I don't have much experience with pickle. I gave it a try and it errored on simple CQ objects. We might be able to get it working with [`__getstate__`](https://docs.python.org/3/library/pickle.html#object.__getstate__) methods on a few CQ classes.